frontend/components/comments/comment_reply.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import {Link} from 'react-router-dom';
+import ProfileButton from '../util/profile_button';
+import TimeAgo from '../util/time_ago';
+import CommentLikesContainer from './comment_likes_container';
+import EditMenu from '../videos/edit-comment-menu';
+import CommentForm from '../videos/comment-form';
+
+class CommentReply extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showEditMenu: false,
+      showEditMenuBtn: false,
+      editComment: false,
+    };
+
+    this.showEditMenuBtn = this.showEditMenuBtn.bind(this);
+    this.hideEditMenuBtn = this.hideEditMenuBtn.bind(this);
+    this.toggleEditMenu = this.toggleEditMenu.bind(this);
+    this.hideEditMenu = this.hideEditMenu.bind(this);
+    this.showEditMenu = this.showEditMenu.bind(this);
+    this.editComment = this.editComment.bind(this);
+    this.deleteComment = this.deleteComment.bind(this);
+    this.updateComment = this.updateComment.bind(this);
+    this.setEditMenuBtn = this.setEditMenuBtn.bind(this);
+    this.cancelEditComment = this.cancelEditComment.bind(this);
+    this.focus = this.focus.bind(this);
+    this.blur = this.blur.bind(this);
+  }
+
+  update(field) {
+    return (e) => {
+      this.setState({ [field]: e.target.value });
+    }
+  }
+
+  editComment() {
+    this.blur();
+    this.setState({ editComment: true });
+  }
+
+  cancelEditComment() {
+    this.setState({ editComment: false });
+  }
+
+  updateComment(formData) {
+    this.props.updateComment(formData);
+    this.setState({ editComment: false });
+  }
+
+  deleteComment() {
+    this.blur();
+    this.props.deleteComment(this.props.comment.id);
+  }
+
+  setEditMenuBtn(node) {
+    this.editMenuBtn = node;
+  }
+
+  focus() {
+    this.editMenuBtn.focus();
+  }
+
+  blur() {
+    this.editMenuBtn.blur();
+  }
+
+  showEditMenuBtn(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ showEditMenuBtn: true });
+  }
+
+  hideEditMenuBtn(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ showEditMenuBtn: false });
+  }
+
+  showEditMenu(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ showEditMenu: true });
+  }
+
+  hideEditMenu(e) {
+    e.preventDefault();
```

Comment:
preventDefault() is to stop the browser from executing its default behavior with form submissions (if I'm not mistaken). you have some extra preventing of default in some of these methods I believe